src/uri/path.rs

```diff
@@ -49,6 +49,11 @@ impl PathAndQuery {
             match URI_CHARS[b as usize] {
                 0 => {
                     if b == b'%' {
+                        // Check if next character is not %
+                        if i + 1 > src.len() && b'%' == src[i + 1] {
```

Comment:
Shouldn't it be: 
```rust
if i + 1 >= src.len() || b'%' == src[i + 1] {
```

Reply:
apparently should have been +2, why `||` OR ?